### PR TITLE
Add size interface to list

### PR DESCRIPTION
**Pull request by quincyle**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
* **Removed** default value for `defaultHeight` property.
* **Updated** `smoke-and-mirror` to consume `listRowHeight` instead of `defaultHeight`.
* **Updated** `frost-checkbox` in `frost-list-item-section`, now it reacts to `size`.
* **Updated** `frost-list-item-container-base` to generate its height based on `listRowHeight`.
 * **Added** new interface `size` that defaults to `medium`.
 * **Added** new computed property `listRowHeight` calculates based on `defaultHeight` and `size`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,8 +60,7 @@ Detailed API and example usage can be found in the sample application in tests/d
 | `Sub Attribute`    | `properties`         | `array`          |       | Array of sortable attributes. eg. [{"label: "foo", "value": "bar"}], This is an attribute on frost-sort component.|
 | `Sub Attribute`    | `onSort`             | `action closure` |       | callback functions user provided to handle sorting.  This is an attribute on frost-sort component.|
 | `Attribute`        | `itemKey`            | `string`         |       | Optional: With itemKey set, item.get(itemKey) will be used for comparision, Else the default item === item comparison used. |
-
-
+| `Attribute`        | `size`               | `string`         |       | Optional: Size can take either "small" or "medium". With size set to "small", list will provide a compressed list item view. This attribute gets ignored when defaultHeight is provided. |
```
Comment (notmessenger) [inline README.md]: Would be helpful to note that size is `medium` by default.

File: addon/components/frost-list.js [hunk 2]
```diff
@@ -114,6 +116,29 @@ export default Component.extend({
...
   },
 
+  @readOnly
+  @computed('defaultHeight', 'size')
+  listRowHeight (defaultHeight, size) {
+    if (isPresent(defaultHeight)) {
+      return defaultHeight
+    }
+
+    switch (size) {
```
Comment (notmessenger) [inline addon/components/frost-list.js]: Have this statement set a value that gets returned - too many `return` statements

Comment (notmessenger) [reply addon/components/frost-list.js]: Would prefer to actually have all of the logic be approached so there is only one `return` for the entire method.

File: tests/dummy/app/pods/size/controller.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import computed, {readOnly} from 'ember-computed-decorators'
+import {sort} from 'ember-frost-sort'
+
+export default Controller.extend({
+
+  // == Dependencies ==========================================================
+
+  // == Properties ============================================================
+
+  expandedItems: A([]),
```
Comment (notmessenger) [inline tests/dummy/app/pods/size/controller.js]: Can be just `A()` (same for all other instances in this file)

**Review by notmessenger — CHANGES_REQUESTED** [review]

File: tests/dummy/app/pods/size/controller.js (hunk 3 above)
Comment (quincyle) [reply tests/dummy/app/pods/size/controller.js]: Good catch, copied from other dummy routes. In order to make this feature PR clean, I'll raise another PR later to clean up the rest of the dummy.

File: tests/dummy/app/pods/size/controller.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/**
```
Comment (notmessenger) [inline tests/dummy/app/pods/size/controller.js]: This is still present.

**Review by notmessenger — CHANGES_REQUESTED** [review]

File: tests/dummy/app/pods/size/controller.js (hunk 4 above)
Comment (quincyle) [reply tests/dummy/app/pods/size/controller.js]: which line you're referring to?

Comment (notmessenger) [reply tests/dummy/app/pods/size/controller.js]: https://github.com/quincyle/ember-frost-list/blob/84ccd52c5bafe328ed9c2c99cff5b01c9d43f4db/tests/dummy/app/pods/size/controller.js#L1-L3

File: README.md [hunk 5]
```diff
@@ -60,13 +60,15 @@ Detailed API and example usage can be found in the sample application in tests/d
...
 | `Attribute`        | `componentKeyNames`  | `hash`           |       | Optional: With `componentKeyNames` set, these key names will be used to identify component key names in `componentKeyNamesForTypes`. Only applicable when using a list with more than one type of item. |
 | `Attribute`        | `itemTypeKey`        | `string`         |       | Optional: With `itemTypeKey` set, it will be used to identify a list item's type. It will also be used for accessing the hash of components within `componentKeyNamesForTypes`. |
 | `Attribute`        | `itemDefinitions`    | `hash`           |       | Optional: A set of components that are to be used in the list as the `item` component. Note that this had to be used in conjunction with `componentKeyNamesForTypes` |
 | `Attribute`        | `itemExpansionDefinitions` | `hash`     |       | Optional: A set of components that are to be used in the list as the `itemExpansion` component. Note that this had to be used in conjunction with `componentKeyNamesForTypes` |
 
 
+>>>>>>> cc90c3107e5d88e81b88119af521d6e4775e8aac
```
Comment (notmessenger) [inline README.md]: Remove

**Review by notmessenger — CHANGES_REQUESTED** [review]

**Review by notmessenger — APPROVED** [review]

**Comment by notmessenger** [comment]
👍

**Comment by quincyle** [comment]
@notmessenger Appreciate for the fast review.

**Pull request merged**